cmd/skuba/cluster/check.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/SUSE/skuba/pkg/skuba"
+	"github.com/rikatz/kubepug/lib"
+	"github.com/rikatz/kubepug/pkg/formatter"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+)
+
+type checkOptions struct {
+	K8sVersion      string
+	ForceDownload   bool
+	APIWalk         bool
+	SwaggerDir      string
+	ShowDescription bool
+}
+
+var (
+	format    string
+	filename  string
+	inputFile string
+)
+
+// NewCheckCmd creates a new `skuba check` cobra command
+func NewCheckCmd() *cobra.Command {
+	checkOptions := &checkOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "check k8s-version=<version> swaggerDir=<directory> --api-walk=<true|fasle>",
+		Short: "Print Check information",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubernetesConfigFlags := genericclioptions.NewConfigFlags(true)
+			kubeAdminFile := skuba.KubeConfigAdminFile()
+			kubernetesConfigFlags.KubeConfig = &kubeAdminFile
+
+			config := lib.Config{
+				K8sVersion:      checkOptions.K8sVersion,
+				ForceDownload:   checkOptions.ForceDownload,
+				APIWalk:         checkOptions.APIWalk,
+				SwaggerDir:      checkOptions.SwaggerDir,
+				ShowDescription: checkOptions.ShowDescription,
+				ConfigFlags:     kubernetesConfigFlags,
+				Input:           inputFile,
+			}
+
+			klog.Infof("Starting Kubepug with configs: %+v", config)
+			kubepug := lib.NewKubepug(config)
+
+			result, err := kubepug.GetDeprecated()
+			if err != nil {
+				return err
+			}
+
+			klog.Info("Starting deprecated objects printing")
+			formatter := formatter.NewFormatter(format)
+			bytes, err := formatter.Output(*result)
+			if err != nil {
+				return err
+			}
+
+			if filename != "" {
+				err = ioutil.WriteFile(filename, bytes, 0644)
+				if err != nil {
+					return err
+				}
+			} else {
+				fmt.Printf("%s", string(bytes))
+			}
+
+			return nil
+		},
+		Args: cobra.MinimumNArgs(1),
+	}
+	cmd.PersistentFlags().BoolVar(&checkOptions.APIWalk, "api-walk", true, "Wether to walk in the whole API, checking if all objects type still exists in the current swagger.json. May be IO intensive to APIServer. Defaults to true")
```

Comment:
typo? `Whether`

Reply:
The cmd flag supports prints out the default value, so we don't have to write `Defaults to true` in the help message manually.

Reply:
> typo? `Whether`

+1

Reply:
> The cmd flag supports prints out the default value, so we don't have to write `Defaults to true` in the help message manually.

Oh I forgot about that. Good catch!